pkg/auth/providers/saml/saml_client.go

```diff
@@ -257,7 +260,12 @@ func (s *Provider) HandleSamlAssertion(w http.ResponseWriter, r *http.Request, a
 		return
 	}
 
-	userPrincipal, groupPrincipals = s.getSamlPrincipals(config, samlData)
+	userPrincipal, groupPrincipals, err = s.getSamlPrincipals(config, samlData)
+	if err != nil {
+		log.Error(err)
+		http.Redirect(w, r, redirectURL+"/login?errorCode=500", http.StatusFound)
+		return
+	}
```

Comment:
Are we redirecting to Rancher UI with errorCode that the UI is translating to an error message?

Reply:
clarified offline.